transforms3d/quaternions.py

```diff
@@ -439,20 +438,37 @@ def quat2axangle(quat, identity_thresh=None):
     >>> quat2axangle([1, 0, 0, 0])
     (array([ 1.,  0.,  0.]), 0.0)
 
+    If any of the quaternion values are not finite, we return a NaN in the
+    angle, and an arbitrary vector:
+
+    >>> quat2axangle([1, np.inf, 0, 0])
+    (array([ 1.,  0.,  0.]), nan)
+
     Notes
     -----
     A quaternion for which x, y, z are all equal to 0, is an identity rotation.
-    In this case we return a 0 angle and an arbitrary vector, here [1, 0, 0]
+    In this case we return a 0 angle and an arbitrary vector, here [1, 0, 0].
+
+    The algorithm allows for quaternions that have not been normalized.
     '''
     w, x, y, z = quat
-    vec = np.asarray([x, y, z])
+    Nq = w * w + x * x + y * y + z * z
+    if not np.isfinite(Nq):
+        return np.array([1.0, 0, 0]), float('nan')
```

Comment:
@alexis-roche - sorry to ask you about this, but, I am hoping to replace the nipy copy of these routines with this package ('transforms3d').

In the nipy version of the code, you have this:

```
    w, x, y, z = quat
    vec = np.asarray([x, y, z])
    if identity_thresh is None:
        try:
            identity_thresh = np.finfo(vec.dtype).eps * 3
        except ValueError:  # integer type
            identity_thresh = _FLOAT_EPS * 3
    len2 = x * x + y * y + z * z
    if len2 < identity_thresh ** 2:
        # if vec is nearly 0,0,0, this is an identity rotation
        return np.array([1.0, 0, 0]), 0.0
    theta = 2 * math.acos(max(min(w, 1), -1))
    if len2 == float('inf'):
        return np.zeros((3,)), theta
    return vec / math.sqrt(len2), theta
```

So, we check for 'inf' in the last three values only, and we don't normalize the quaternion.   Thinking about it, I think we should normalize the quaternion, and with the big values previously producing `inf` in the last three values, the normalization will always fail with non-finite values.  Do you think my angle == `nan` approach makes sense?   Would it work OK for the registration code too?